server/service/handler.go

```diff
@@ -333,6 +333,13 @@ func attachFleetAPIRoutes(r *mux.Router, svc fleet.Service, config config.FleetC
 	ue.GET("/api/_version_/fleet/status/result_store", statusResultStoreEndpoint, nil)
 	ue.GET("/api/_version_/fleet/status/live_query", statusLiveQueryEndpoint, nil)
 
+	// device-authenticated endpoints
+	de := newDeviceAuthenticatedEndpointer(svc, logger, opts, r, "v1")
+	de.GET("/api/_version_/fleet/device/{token}", getDeviceHostEndpoint, getDeviceHostRequest{})
+	de.POST("/api/_version_/fleet/device/{token}/refetch", refetchDeviceHostEndpoint, refetchDeviceHostRequest{})
+	de.GET("/api/_version_/fleet/device/{token}/device_mapping", listDeviceHostDeviceMappingEndpoint, listDeviceHostDeviceMappingRequest{})
+	de.GET("/api/_version_/fleet/device/{token}/macadmins", getDeviceMacadminsDataEndpoint, getDeviceMacadminsDataRequest{})
```

Comment:
Having the auth token in the path is not secure, as it will be logged in various places. Given that our endpoints are read-only and don't expose anything critical AFAIK, it might be fine, but just wanted to point it out in case it hasn't been fully considered (the token is not super secured on the host anyway, so it should always be treated as a weak form of auth and should have very limited access).

Reply:
If you are able to read the logs you can probably see everything you can see with these endpoints.

That said, you do have a good point, and we should treat it carefully.